### ISSUE TITLE
Add the Context type

### ISSUE THREAD

**Issue by achille-roussel** [segmentio/analytics-go#59]
@f2prateek 

The sepcs says the context field is an object with a specific structure (unlike properties or traits which are free-form objects).

How about defining an actual struct type to make the code more self-documenting on what can be set?

I guess my main question is, could this be too restrictive in some cases or is there any undocumented parts of the context that our API actually supports?

https://segment.com/docs/spec/common/#context

**Comment by f2prateek** [comment]
Oh wow I never realized context was limited to just what we documented. +1 to making it a specific type (we'd probably want to make all the fields it defines their own types too).

**Comment by achille-roussel** [comment]
Cool, when I'll get to implementing this I'll probably have more questions for you then because some of the examples show things that aren't documented for the fields of Context.

**Comment by achille-roussel** [comment]
Which by the way, there are no `action` field set in analytics-go... I guess our backends support either `action` or `type` or the document is out of date?

**Comment by paddie** [comment]
Just chiming in here. This was massively confusing while doing our stuff, so landing on either action or type and having the docs reflect this would be great.

@achille-roussel I actually ended up creating a Context struct myself, precisely like you are proposing. The reason I didn't want to introduce it, was that I figured you guys were keeping it generic, to be future-compliant if you introduced a new field.
